Product/Production/Common/CONNECTCoreLib/src/test/java/gov/hhs/fha/nhinc/transform/policy/XDRPolicyTransformHelperTest.java

```diff
@@ -0,0 +1,227 @@
+
+		verifycheckPolicy_Request(result, direction);
+
+	}
+
+	@Test
+	public void testTransformXDRResponseToCheckPolicy() throws Exception {
+
+		XDRResponseEventType eventA = new XDRResponseEventType();
+
+		HomeCommunityType hct = new HomeCommunityType();
+		hct.setHomeCommunityId(TEST_HC_VAL);
+		hct.setName("bcd");
+		hct.setDescription("jkl");
+		AssertionType assertionTypeTest = new AssertionType();
+		assertionTypeTest.setMessageId(MESSAGE_ID);
+		SlotType1 slotType1 = new SlotType1();
+		slotType1.setName("$XDSDocumentEntryPatientId");
+
+		ValueListType valueListType = new ValueListType();
+		valueListType.getValue().add("1111.4444^^^&amp;26.489.22&amp;ISO");
+		slotType1.setValueList(valueListType);
+
+		PersonNameType pnt = new PersonNameType();
+		pnt.setFullName("abc");
+		assertionTypeTest.setPersonName(pnt);
+		SlotListType rsl = new SlotListType();
+
+		rsl.getSlot().add(slotType1);
+
+		SubmitObjectsRequest sor = new SubmitObjectsRequest();
+		sor.setId("100");
+		sor.setRequestSlotList(rsl);
+
+		RegistryResponseType prdsrt = new RegistryResponseType();
+
+		prdsrt.setResponseSlotList(rsl);
+		prdsrt.setRequestId("1235");
+		RegistryObjectListType rolt = new RegistryObjectListType();
+
+		sor.setRegistryObjectList(rolt);
+
+		XDRResponseMessageType xmt = new XDRResponseMessageType();
+		eventA.setMessage(xmt);
+		xmt.setAssertion(assertionTypeTest);
+		xmt.setRegistryResponse(prdsrt);
+
+		assertionTypeTest.setHomeCommunity(hct);
+		assertionTypeTest.setSSN("170112756");
+		eventA.setSendingHomeCommunity(hct);
+		String direction = "inbound";
+		eventA.setDirection(direction);
+		eventA.getMessage().setAssertion(assertionTypeTest);
+
+		RequestType request = new RequestType();
+		SubjectHelper subjHelp = new SubjectHelper();
+		SubjectType subject = subjHelp.subjectFactory(hct, assertionTypeTest);
+		request.getSubject().add(subject);
+		CheckPolicyRequestType result = xdrTransformHelper.transformXDRResponseToCheckPolicy(eventA);
+
+		verifycheckPolicy_Request(result, direction);
+
+	}
+
+	private void verifycheckPolicy_Request(CheckPolicyRequestType actualCheckPolicy, String direction) {
+
+		assertNotNull(actualCheckPolicy);
+
+		AssertionType assertTypeResult = actualCheckPolicy.getAssertion();
+
+		assertEquals("messageId should not change", MESSAGE_ID, assertTypeResult.getMessageId());
+		SubjectType subjectType = actualCheckPolicy.getRequest().getSubject().get(0);
+		AttributeType attributeType = subjectType.getAttribute().get(1);
+		assertEquals("http://www.hhs.gov/healthit/nhin#HomeCommunityId", attributeType.getAttributeId());
+		assertEquals("http://www.w3.org/2001/XMLSchema#anyURI", attributeType.getDataType());
+		assertEquals(TEST_HC_VAL, attributeType.getAttributeValue().get(0).getContent().get(0));
+
+	}
+
+	@Test
+	public void testNull_Event() {
+
+		XDRResponseEventType eventA = new XDRResponseEventType();
+		XDREventType eventB = new XDREventType();
```

Comment:
Following up on the comment from @mhpnguyen - is there any value in declaring eventA and eventB, as opposed to using the null directly?  e.g.,

`assertNull(xdrTransformHelper.transformXDRResponseToCheckPolicy(null));`